src/plugins/controls/public/options_list/components/options_list_editor_options.tsx

```diff
@@ -6,43 +6,96 @@
  * Side Public License, v 1.
  */
 
-import React, { useState } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
 import {
   EuiFlexGroup,
   EuiFlexItem,
+  EuiForm,
   EuiFormRow,
   EuiIconTip,
+  EuiSuperSelectOption,
+  EuiSpacer,
+  EuiSuperSelect,
   EuiSwitch,
   EuiSwitchEvent,
+  EuiButtonGroup,
+  toSentenceCase,
+  Direction,
 } from '@elastic/eui';
 import { css } from '@emotion/react';
 
+import {
+  getCompatibleSortingTypes,
+  sortDirections,
+  DEFAULT_SORT,
+  SortBy,
+} from '../../../common/options_list/suggestions_sorting';
 import { OptionsListStrings } from './options_list_strings';
 import { ControlEditorProps, OptionsListEmbeddableInput } from '../..';
+
 interface OptionsListEditorState {
-  singleSelect?: boolean;
+  sortDirection: Direction;
   runPastTimeout?: boolean;
+  singleSelect?: boolean;
   hideExclude?: boolean;
   hideExists?: boolean;
+  hideSort?: boolean;
+  sortBy: SortBy;
 }
 
 interface SwitchProps {
   checked: boolean;
   onChange: (event: EuiSwitchEvent) => void;
 }
 
+type SortItem = EuiSuperSelectOption<SortBy>;
+
 export const OptionsListEditorOptions = ({
   initialInput,
   onChange,
+  fieldType,
 }: ControlEditorProps<OptionsListEmbeddableInput>) => {
   const [state, setState] = useState<OptionsListEditorState>({
-    singleSelect: initialInput?.singleSelect,
+    sortDirection: initialInput?.sort?.direction ?? DEFAULT_SORT.direction,
+    sortBy: initialInput?.sort?.by ?? DEFAULT_SORT.by,
     runPastTimeout: initialInput?.runPastTimeout,
+    singleSelect: initialInput?.singleSelect,
     hideExclude: initialInput?.hideExclude,
     hideExists: initialInput?.hideExists,
+    hideSort: initialInput?.hideSort,
   });
 
+  useEffect(() => {
+    // when field type changes, ensure that the selected sort type is still valid
```

Comment:
Nice. I did a lot of testing where I would select a sort, then change the field type looking for bugs before I saw this `useEffect`. Very clean! :fire: